opencog/query/InitiateSearchCB.cc

```diff
@@ -620,40 +620,42 @@ bool InitiateSearchCB::variable_search(PatternMatchEngine *pme)
 		const std::set<Type>& typeset = tit->second;
 		LAZY_LOG_FINE << "Type-restictions set size = "
 		              << typeset.size();
+
+		// Calculate the total number of atoms of typeset
+		size_t num = 0;
 		for (Type t : typeset)
+			num += (size_t) _as->get_num_atoms_of_type(t);
+
+		LAZY_LOG_FINE << var->toString() << "has "
+		              << num << " atoms in the atomspace";
+
+		if (0 < num and num < count)
 		{
-			size_t num = (size_t) _as->get_num_atoms_of_type(t);
-			LAZY_LOG_FINE << "Type = "
-			              << _classserver.getTypeName(t) << " has "
-			              << num << " atoms in the atomspace";
-			if (0 < num and num < count)
+			for (const Handle& cl : clauses)
 			{
-				for (const Handle& cl : clauses)
+				// Evaluatables dont' exist in the atomspace, in general.
+				// Cannot start a search with them.
+				if (0 < _pattern->evaluatable_holders.count(cl)) continue;
```

Comment:
There would be find, but since it's a set (as opposed to a multiset), count should be just as fast.